### ISSUE TITLE
How can I get IPs through  cf API 

### ISSUE THREAD

**Issue by zhengduoyu** [cloudfoundry/cloud_controller_ng#1378]
Hi friends
   I want to get three fields of information  through  cf API :  CF_INSTANCE_IP，CF_INSTANCE_INTERNAL_IP and CF_INSTANCE_PORT  .But I didn't get a reference from helpful resources.   How can I get these fields

**Comment by zhengduoyu** [comment]
I can get these fiels by instruction : cf ssh $appName;     env | grep INATANCE .    But I want to get those by api

**Comment by sethboyles** [comment]
Hmm, I don't believe we provide those via the API.  What use case do you have that would require those variables?

**Comment by tcdowney** [comment]
@zhengduoyu 

Does this endpoint do what you need?
http://v3-apidocs.cloudfoundry.org/version/3.72.0/index.html#get-stats-for-a-process

One gotcha with it is that the ports it displays can be misleading if you have [route integrity enabled](https://docs.cloudfoundry.org/concepts/http-routing.html#with-tls), though. See this story for more information on improving that experience: https://www.pivotaltracker.com/story/show/166934752

**Comment by tcdowney** [comment]
@zhengduoyu just checking in again, does the `/v3/processes/:process-guid/stats` endpoint[0] give you what you need? It has the IP of the host that is running the app container in its response.

```
{
  "resources": [
    {
      "type": "web",
      "index": 0,
      "state": "RUNNING",
      "usage": {
        "time": "2016-03-23T23:17:30.476314154Z",
        "cpu": 0.00038711029163348665,
        "mem": 19177472,
        "disk": 69705728
      },
      "host": "10.244.16.10",
      "instance_ports": [
        {
          "external": 64546,
          "internal": 8080
        }
      ],
      "uptime": 9042,
      "mem_quota": 268435456,
      "disk_quota": 1073741824,
      "fds_quota": 16384,
      "isolation_segment": "example_iso_segment",
      "details": null
    }
  ]
}
```

[0] http://v3-apidocs.cloudfoundry.org/version/3.74.0/index.html#get-stats-for-a-process

**Comment by cwlbraa** [comment]
https://www.pivotaltracker.com/story/show/166934752 is related to this

**Comment by sethboyles** [comment]
@zhengduoyu I am closing due to inactivity.  Please reopen if the above endpoints do not provide the information you need.

**Comment by ameowlia** [comment]
I am writing a new CAT for readiness healthcheck today, and I was looking for exactly this feature, so I could grab the CF_INSTANCE_INTERNAL_IP variable. It would be nicer if I could grab it from the API vs the cf cli.

**Comment by sethboyles** [comment]
@ameowlia how does the CLI get it?

**Comment by tcdowney** [comment]
I think the [net_info](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/lib/cloud_controller/diego/reporters/instances_stats_reporter.rb#L38) probably has it on there. I might poke at that real quick and see if it is actually there.

**Comment by tcdowney** [comment]
@sethboyles looks like you could do something like this
https://github.com/cloudfoundry/cloud_controller_ng/pull/3378

If this is a change y'all are actually looking for I'd be happy to make a real PR.

**Comment by moleske** [comment]
resolved with https://github.com/cloudfoundry/cloud_controller_ng/pull/3378 and released in [capi 1.162.0](https://github.com/cloudfoundry/capi-release/releases/tag/1.162.0)
